### DISCUSSION TITLE
Rocky 8.10 installation

### DISCUSSION THREAD

**Discussion by thomase00** [Xpra-org/xpra#4399] [Q&A] [answered]
In my organization, I'm not allowed super-user permissions for my VDI, but I can a request that an admin install packages. I pointed to the installation instructions here https://github.com/Xpra-org/xpra/wiki/Download#-for-rpm-distributions, which they followed apparently, but I only have version 4.4, while it looks like Rocky Linux 8.x should support the latest version (currently 6.2).

Anyone have any idea why I wouldn't have the latest version?

**Comment by totaam** [comment]
> Anyone have any idea why I wouldn't have the latest version?

Without the log of the installation attempt, no.
My guess is that the gpg key was not imported properly, so you end up with the downstream version instead of the one from xpra.org.

**Comment by thomase00** [comment]
Ok, I got them to install the latest, but I'm observing a few issues that I
don't recall running into in the past, or if I did, what the solution was.

1.) X clients running on hosts other than the one running the xpra server
can't open the xpra display, EVEN after running 'xhost +' on the xpra
server.

2.) avif and vp9 codecs are missing on the server according to session info

3.) The "Graphs" tab in session info is not showing anything.

4.) Before I got the admin to install the latest xpra packages, I did my
own install with the following steps:

   - Download all the latest rpms from
   https://xpra.org/dists/rockylinux/8/x86_64/.
   - Extract all files to my own user-writable directory.
   - set PATH, PYTHONPATH, LD_LIBRARY_PATH, AND MANPATH env variables to
   point to my local "install".

From what I can see, this also works, with the exception that X application
taskbar icons (on my Windows 10 client) are displaying the generic xpra 'X'
icon rather than the appropriate icon for the application. Not sure what is
different in my local install that would explain this.

**Comment by totaam** [comment]
> X clients running on hosts other than the one running the xpra server
can't open the xpra display, EVEN after running 'xhost +' on the xpra
server.

I have never seen that for seamless or desktop sessions.
Something must be wrong with the paths, perhaps `$XDG_RUNTIME_DIR` and friends.

> avif and vp9 codecs are missing on the server according to session info

Bad installation.

> The "Graphs" tab in session info is not showing anything.

Could be a bug, very likely a different issue.

> From what I can see, this also works, with the exception that X application
taskbar icons 

There is an open ticket to do with system tray icons - which could be the same issue: #4401

**Comment by thomase00** [comment]
I haven't found a solution for X clients on other hosts not being able to
open the display. xpra is unusable for me unless I solve this because most
X client applications are running on different hosts via lsf.

Is it the Xorg process itself that determines which X clients can and
cannot connect to it? I'm specifying a custom Xorg command with the
`--xvfb` switch because I want to allow tcp connections (the default has
nolisten tcp).

Could that be related?

Here's my xvfb command:

```
/usr/libexec/Xorg -noreset -novtswitch +extension GLX +extension RANDR \
    +extension RENDER -logfile ${HOME}/.xpra/$1/Xorg.log -config \
    ${HOME}/.config/xpra/xorg.conf
```

**Comment by totaam** [comment]
Have you tried allowing access using `xhost` ?
xpra will add an xauth entry:
https://github.com/Xpra-org/xpra/blob/754774ba0a2b19821c59e240672c32f4bb106fcc/xpra/scripts/server.py#L1203-L1248

Since your installation is a bit different, maybe the `XAUTHORITY` path used is unexpected.
I believe that starting your server with `-d server` would show some `xauth` related debug logging, including the path used:
https://github.com/Xpra-org/xpra/blob/754774ba0a2b19821c59e240672c32f4bb106fcc/xpra/scripts/server.py#L1262

I don't think you would need this, but here it is for completeness:
https://github.com/Xpra-org/xpra/blob/754774ba0a2b19821c59e240672c32f4bb106fcc/xpra/scripts/server.py#L63

**Comment by thomase00** [comment]
I've tried "xhost +" if that is what you are asking.

**Comment by totaam** [comment]
If it was me, I would:
* verify that the Xorg process is listening on a TCP socket
* verify that I can connect to it using `telnet`, then an `xterm`
* verify that my firewall isn't blocking this port to the outside
* verify that I can connect to it from the outside

This has nothing to do with xpra.

**Comment by thomase00** [comment, marked as answer]
Figured it out.

I guess the version of Xorg on Rocky8 defaults to NOT listening on TCP
ports. Therefore, rather than removing '-nolisten tcp' from the Xorg
command line, I changed it to '-listen tcp'.
